File: 2_e_validate_subsequence.js
function isValidSubsequence(array, sequence) {
  // Write your code here.
  // 	O(n) time n is the length of array, O(1) space
  let seqIdx = 0;
  let arrIdx = 0;
  while (arrIdx < array.length && seqIdx < sequence.length) {
    if (sequence[seqIdx] === array[arrIdx]) {
      seqIdx++;
    }
    arrIdx++;
  }
  return seqIdx === sequence.length;
}

// solution 2

function isValidSubsequence(array, sequence) {
  // Write your code here.
  let seqIdx = 0;
  for (let n of array) {
    if (seqIdx === sequence.length) {
      break;
    }
    if (sequence[seqIdx] === n) {
      seqIdx++;
    }
  }
  return sequence.length === seqIdx;
}

File: 2_e_validate_subsequence.py
def isValidSubsequence(array, sequence):
    # Write your code here.
    #O(n) O(1)
  i = 0
  j = 0
  while i < len(array) and j < len(sequence):
    if array[i] == sequence[j]:
      j += 1
    i += 1
  return j == len(sequence)

#second method
def isValidSubsequence(array, sequence):
  i = 0
  for n in array:
    if n == sequence[i]:
      i += 1
    if i == len(sequence):
      break
  return i == len(sequence)